Add Interactions GameRules

FirstMegaGame4 opened the pull request:
Add `canEntityInteractWithBlock` GameRule
Add `canEntityInteractWithEntity` GameRule
Add `canEntityInteractWithWorld` GameRule
Close #21

FirstMegaGame4 committed:
Add `canPlayerInteractWithBlock` GameRule
Add `canPlayerInteractWithEntity` GameRule
Add `canPlayerInteractWithWorld` GameRule
(these gamerules are not finished yet)

Max094Reikeb commented on src/main/java/net/reikeb/notenoughgamerules/Gamerules.java:
```diff
@@ -42,6 +45,9 @@ public static void setupGamerules() {
         ANVIL_DAMAGE = GameRuleRegistry.register("anvilDamage", GameRules.Category.PLAYER, GameRuleFactory.createBooleanRule(true));
         CAN_COPPER_OXIDE = GameRuleRegistry.register("canCopperOxide", GameRules.Category.UPDATES, GameRuleFactory.createBooleanRule(true));
         CAN_HURT_PET_MOBS = GameRuleRegistry.register("canHurtPetMobs", GameRules.Category.MOBS, GameRuleFactory.createBooleanRule(true));
+        CAN_PLAYER_INTERACT_WITH_BLOCK = GameRuleRegistry.register("canPlayerInteractWithAction", GameRules.Category.PLAYER, GameRuleFactory.createBooleanRule(true));
```
Please rename the gamerule ID to `canPlayerInteractWithBlock`

FirstMegaGame4 replied:
oh i forgot that

Max094Reikeb commented on src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/AbstractButtonBlockMixin.java:
```diff
@@ -0,0 +1,23 @@
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
```
Don't forget to organise imports

FirstMegaGame4 replied:
yes I will do that later when the pr is ready to be merged

Max094Reikeb commented on src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/AbstractPressurePlateBlockMixin.java:
```diff
@@ -0,0 +1,20 @@
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
```
Same here

Max094Reikeb commented on src/main/java/net/reikeb/notenoughgamerules/mixin/blocks/LeverBlockMixin.java:
```diff
@@ -0,0 +1,23 @@
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
```
Same here

FirstMegaGame4 committed:
Rename GameRule `canPlayerInteractWithBlock` to `canEntityInteractWithBlock` and Finish it.
Rename GameRule `canPlayerInteractWithEntity` to `canEntityInteractWithEntity` and Finish it.
Rename GameRule `canPlayerInteractWithEntity` to `canEntityInteractWithWorld`and Finish it.

FirstMegaGame4 committed:
Reorganize Imports

FirstMegaGame4 committed:
Add to en_us.json

FirstMegaGame4 commented:
Fix #21

Max094Reikeb commented on src/main/resources/assets/not_enough_gamerules/lang/en_us.json:
```diff
@@ -6,6 +6,9 @@
   "gamerule.canCopperOxide.description": "Defines if normal copper can oxidize",
+  "gamerule.canEntityInteractWithBlock": "Can Entity Interact With Blocks",
+  "gamerule.canEntityInteractWithEntity": "Can Entity Interact With Another Entities",
+  "gamerule.canEntityInteractWithWorld": "Can Entity Interact With World",
```
No need to make every work start with a cap
`Can entity interact with blocks` is good enough...

FirstMegaGame4 committed:
Change en_us.json

Max094Reikeb commented on src/main/resources/assets/not_enough_gamerules/lang/en_us.json:
```diff
@@ -6,6 +6,9 @@
   "gamerule.canCopperOxide": "Can copper oxide",
   "gamerule.canCopperOxide.description": "Defines if normal copper can oxidize",
+  "gamerule.canEntityInteractWithBlock": "Can entity interact with blocks",
+  "gamerule.canEntityInteractWithEntity": "Can entity interact with another entities",
```
Not very English, either "interact with other entities", or "interact with another entity", but can't be both. You need to choose one.

FirstMegaGame4 replied:
I realized that and was going to make a commit to change that

FirstMegaGame4 committed:
Rename `canEntityInteractWithEntity` to `canEntityInteractWithEntities`
